flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -165,20 +167,10 @@ class ReachCollector {
       popFn(rop);
       return;
     }
-    if (auto rop = mlir::dyn_cast<IterWhileOp>(op)) {
```

Comment:
What was the argument for not handling these loops again?

Reply:
I think I discussed with Jean and apparently the lowering does not produce array ops inside a IterWhileOp. We can keep it if we have have a test case.